backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskSnapshotsResource.java

```diff
@@ -38,11 +39,16 @@ protected DiskSnapshots mapCollection(List<org.ovirt.engine.core.common.business
         for (org.ovirt.engine.core.common.businessentities.storage.Disk disk : entities) {
             DiskSnapshot diskSnapshot = getMapper(org.ovirt.engine.core.common.businessentities.storage.Disk.class, DiskSnapshot.class).map(disk, null);
             diskSnapshot.setDisk(new Disk());
-            diskSnapshot.getDisk().setId(this.diskId.toString());
+            diskSnapshot.getDisk().setId(disk.getId().toString());
             collection.getDiskSnapshots().add(addLinks(populate(diskSnapshot, disk), Disk.class));
-            diskSnapshot.setHref(buildHref(diskId.toString(), diskSnapshot.getId().toString()));
-            if (diskSnapshot.getParent() != null) {
-                diskSnapshot.getParent().setHref(buildParentHref(diskId.toString(), false));
+            diskSnapshot.setHref(buildHref(disk.getId().toString(), diskSnapshot.getId()));
+            if ( (((DiskImage) disk).getImage().getParentDiskId()) != null) {
```

Comment:
IMHO cleaner to define a local variable before the `if`:
`Guid parentDiskId = (((DiskImage) disk).getImage().getParentDiskId());`